frontend/app/components/Home.tsx

```diff
@@ -25,34 +28,41 @@ const filterTools = (tools: Tool[], filter: string): Tool[] => {
 };
 
 function Home () {
-  const [tools, setTools] = useState<null | Tool[]>(null);
+  const [tools, setTools] = useState<undefined | Tool[]>(undefined);
+
+  const { isLoading: getToolsLoading, error: getToolsError } = useQuery('getTools', api.getTools, {
+    onSuccess: (data) => setTools(data)
+  });
```

Comment:
In past React development, I've used a custom hook to manage loading and error state, e.g. `usePromise` in ROHQ and CCM. The TypeScript support though was always a little lacking, and I kinda figured this was probably a solved problem. I spent a significant chunk of time last week experimenting with different libraries, including `react-async` and react-async-hook`. The APIs were pretty similar, but I finally settled on [`react-query`](https://github.com/TanStack/query) since it had everything I wanted (good types, `onSuccess` hooks, large active user base, and other nice fetching features). It might be a little heavier than we need, but I think the advantages outweigh that.

Reply:
Looks like a good, well supported library